### DISCUSSION TITLE
IDAKLUsolver run failed ”“KLU” is not installed

### DISCUSSION THREAD

**Discussion by benshang1997** [pybamm-team/PyBaMM#4350] [Q&A]
Hi everyone,

I am planning to use IDAKLUSolver to simulate a long degradation expriment with a termination stop suggested by other disgussion.

I install pybamm[jax] successfully but meet some issues as follow when i run:

Traceback (most recent call last):
  File "D:\OneDrive - University of Surrey\ai4battery_uos_whu\codes\ai4battery\My_code_for_fast_charging\new_code_0521\just_test\degradation_battery_test.py", line 35, in <module>
    safe_solver = pybamm.IDAKLUSolver()
                  ^^^^^^^^^^^^^^^^^^^^^
  File "D:\OneDrive - University of Surrey\ai4battery_uos_whu\codes\ai4battery\pybamm\solvers\idaklu_solver.py", line 116, in __init__
    raise ImportError("KLU is not installed")
ImportError: KLU is not installed

I Install KLU by using pip and it says the install is successful but the issue is still here.

how should I do to solve it ? 
many thanks

**Comment by abhishek-appana** [comment]
Hi @benshang1997 ,

Are you using WSL or installing it directly on Windows? I was able to install it and use it without any errors via WSL. Did not try the windows installation

@agriyakhetarpal has given detailed instructions for both installation procedures in this [discussion](https://github.com/pybamm-team/PyBaMM/discussions/4296)

**Reply by benshang1997** [reply]
Hi @abhishek-appana ， I install it on windows, so I can only use this solver on WSL?

**Reply by agriyakhetarpal** [reply]
Hi @benshang1997, you should be able to compile the IDAKLU solver on both platforms, with varying levels of difficulty.

**Reply by benshang1997** [reply]
thank you very much , it is folded up by the system and I just found the detailed instruction, I will try
